utils/compare/graph.py

```diff
@@ -0,0 +1,306 @@
+import io
+import pprint
+from json import loads
+import pandas as pd
+from search import ElasticService
+import numpy as np
+
+def graph(oMetrics, nMetrics):
+    metrics = []
+    x=[]
+    y=[]
+    for index, row in oMetrics.iterrows():
+        test = "{}-{}".format(row['profile'], row['messageSize'])
+        value = "{}".format(row['throughput'])
+        x.append(value)
+        y.append(test)
+    old = {'y' : x,
+        'x' : y,
+        'name' : 'Previous results average',
+        'type' : 'bar',
+        'orientation' : 'v'}
+    x=[]
+    y=[]
+    for index, row in nMetrics.iterrows():
+        test = "{}-{}".format(row['profile'], row['messageSize'])
+        value = "{}".format(row['throughput'])
+        x.append(value)
+        y.append(test)
+    new = {'y' : x,
+        'x' : y,
+        'name' : 'Current results average',
+        'type' : 'bar',
+        'orientation' : 'v'}
+    metrics.append(old)
+    metrics.append(new)
+
+    return metrics
+
+
+def get_final_graph_metrics(metric_of_interest, divider,previous_data, current_data): 
+    x=[]
+    y=[]
+    for o, row in previous_data.iterrows():
+
+        test = metric_of_interest
+        value = row[metric_of_interest]
+        x.append(value/divider)
+        
+        y.append(test)
+    #print('previous x ' + str(x))
+    old = {'y' : x,
+        'x' : y,
+        'name' : 'Previous results ' + str(metric_of_interest),
+        'type' : 'bar',
+        'orientation' : 'v'}
+    x=[]
+    y=[]
+    for n, row in current_data.iterrows():
+        test = metric_of_interest
+        value = row[metric_of_interest]
+        x.append(value/divider)
+        y.append(test)
+    #print('final x ' + str(x))
+    new = {'y' : x,
+        'x' : y,
+        'name' : 'Current results ' + str(metric_of_interest),
+        'type' : 'bar',
+        'orientation' : 'v'}
+    metrics = []
+    metrics.append(old)
+    metrics.append(new)
+
+    return metrics
+
+def pd_merge(previous_data, current_data, columns):
+    pd_merged = current_data.merge(previous_data, how='outer', on=columns[:-1])
+    return pd_merged
+
+def find_result(row, tolerancy): 
+
+    if np.isinf(row['difference']): 
+        return "Pass"
+    # Fail if value is a more negative than tolerancy
```

Comment:
`README.md` line#16 says better performance in negative case, but here we are marking it as Fail 🤔

Reply:
I had just switched the way the dataframe was set up switching the percent values, will update to be sure they match